airbyte-featureflag/src/main/kotlin/config/Factory.kt

```diff
@@ -20,22 +19,22 @@ internal const val CONFIG_OSS_KEY = "airbyte.feature-flag.path"
 
 @Factory
 class Factory {
-    @Requirements(
-        Requires(property = CONFIG_LD_KEY),
-        Requires(missingProperty = CONFIG_OSS_KEY),
-    )
+
+    @Requires(property = CONFIG_LD_KEY)
     @Singleton
-    fun Cloud(@Property(name = CONFIG_LD_KEY) apiKey: String): FeatureFlagClient {
+    fun LaunchDarklyClient(@Property(name = CONFIG_LD_KEY) apiKey: String): FeatureFlagClient {
         val client = LDClient(apiKey)
         return LaunchDarklyClient(client)
     }
 
-    @Requirements(
-        Requires(property = CONFIG_OSS_KEY),
-        Requires(missingProperty = CONFIG_LD_KEY),
-    )
-    fun Platform(@Property(name = CONFIG_OSS_KEY) configPath: String): FeatureFlagClient {
-        val path = Path.of(configPath)
+    @Requires(missingProperty = CONFIG_LD_KEY)
+    @Singleton
+    fun ConfigFileClient(@Property(name = CONFIG_OSS_KEY) configPath: String): FeatureFlagClient {
+        val path: Path? = if (configPath.isNotBlank()) {
```

Comment:
Out of curiosity, Kotlin doesn't have a more succinct way to represent this type of if/else?  Or, are you just avoiding using a ternary or elvis operator for stylistic/readability reasons?